tools/integration_tests/util/setup/setup.go

```diff
@@ -169,17 +174,12 @@ func UnMount() error {
 func executeTest(m *testing.M) (successCode int) {
 	successCode = m.Run()
 
-	os.RemoveAll(mntDir)
-
 	return successCode
 }
 
 func ExecuteTestForFlagsSet(flags []string, m *testing.M) (successCode int) {
 	var err error
 
-	// Clean the mountedDirectory before running any tests.
-	os.RemoveAll(mntDir)
```

Comment:
Removed this because for readonly tests it is throwing an error -> can't delete from readonly mount.
We are already cleaning mountedDirectory in most of the tests before testing, so I added it in each tests.

Reply:
instead of making changes in every file, isn't it better to 
(a) remove os.RemoveAll from here, create a wrapper function over ExecuteTestForFlagsSet which does os.RemoveAll and then call ExecuteTestForFlagsSet. After that use wrapper function where you want to do cleanup and use ExecuteTestForFlagsSet where don't.
OR
(b) Pass a flag to ExecuteTestForFlagsSet which do cleanup only when true.